ros/src/twist_controller/dbw_node.py

```diff
@@ -133,25 +135,43 @@ def maybe_reset_controller(self):
             rospy.loginfo('Resetting Controller')
 
     def publish(self, throttle, brake, steer):
-        # Do not ever publish throttle and brake at the same time!
-        if brake > 0.0:
-            bcmd = BrakeCmd()
-            bcmd.enable = True
-            bcmd.pedal_cmd_type = BrakeCmd.CMD_TORQUE
-            bcmd.pedal_cmd = brake
-            self.brake_pub.publish(bcmd)
-        else:
-            tcmd = ThrottleCmd()
-            tcmd.enable = True
-            tcmd.pedal_cmd_type = ThrottleCmd.CMD_PERCENT
-            tcmd.pedal_cmd = throttle
-            self.throttle_pub.publish(tcmd)
-
+        # Create brake command
+        bcmd = BrakeCmd()
+        bcmd.enable = True
+        bcmd.pedal_cmd_type = BrakeCmd.CMD_TORQUE
+        bcmd.pedal_cmd = brake
+
+        # Create throttle command
+        tcmd = ThrottleCmd()
+        tcmd.enable = True
+        tcmd.pedal_cmd_type = ThrottleCmd.CMD_PERCENT
+        tcmd.pedal_cmd = throttle
+
+        # Create steering command
         scmd = SteeringCmd()
         scmd.enable = True
         scmd.steering_wheel_angle_cmd = steer
+
+        # Publish
+        # NOTE: do not publish throttle and brake at the same time,
+        # unless we switch from one to the other (to prevent the simulator
+        # from keeping the last value)
+        action = 'brake' if brake > 0.0 else 'throttle'
+
+        if action != self.last_action:
```

Comment:
What is the simulator's behaviour when we change from one to another? Wouldn't that cause some weird behaviour?

Reply:
I expect that the simulator applies the commands that we send, until we update those commands. If you tell it to accelerate 100% it will do so until you publish a new throttle. With this change, we make sure that acceleration = 0.0 when braking and viceversa.
Why would it cause a weird behavior?

Reply:
Ah, ok - I see now. I thought that it behaves weird whenever we send both of them. I missed that we want to set the other one to 0, if we switch between them.

Reply:
Yes indeed it behaves weirdly sending both. However my hope is that switching brake/throttle happens rather rarely - otherwise we should fix that in the `twist_controller`.